### ISSUE TITLE
Generating XML 

### ISSUE THREAD

**Issue by cameronwilliams** [philippe44/AirConnect#475]
Running aircast-linux-arm to generate an xml file, I get a segmentation fault. Running it normally it runs fine. Is there something I’m doing wrong?

./aircast-linux-arm -i aircast.xml
[08:02:43.955437] main:914 Starting aircast version: v1.1.7 (May 20 2023 @ 06:21:52)
[08:02:43.955811] main:921 no config file, using defaults
[08:02:43.956302] Start:681 Binding to 192.168.4.48
[08:02:43.974756] Start:706 Starting pico HTTP server on port 58285
[08:02:43.976010] mDNSsearchCallback:533 Updating configuration aircast.xml
Segmentation fault

Vs

./aircast-linux-arm
[07:53:44.196518] main:914 Starting aircast version: v1.1.7 (May 20 2023 @ 06:21:52)
[07:53:44.196914] main:921 no config file, using defaults
[07:53:44.197448] Start:681 Binding to 192.168.4.48
[07:53:44.215568] Start:706 Starting pico HTTP server on port 47801
[07:53:44.219374] AddCastDevice:632 [0xefda98]: creating MAC
[07:53:44.219478] AddCastDevice:646 [0xefda98]: adding renderer (Sleeping Cave) with mac CCCCFB478FA3
[07:53:44.296375] AddCastDevice:632 [0xefe410]: creating MAC
[07:53:44.296467] AddCastDevice:646 [0xefe410]: adding renderer (Play Den) with mac CCCC3A4B783F

**Comment by philippe44** [comment]
You might miss writing rights in the directory it tries to write

**Comment by cameronwilliams** [comment]
You are correct! I tried running the command with sudo and it had no problems, so it was a permission issue. Thanks again.
